### PR TITLE
Update inspectcode version used in CI

### PR DESCRIPTION
**Pull request by smoogipoo**

### CODE REVIEW

**Comment by smoogipoo** [comment]
So I've changed everything to use asserts, just going to say ahead of time that I wholeheartedly disagree with this. Asserts like this rub me the long way, and I don't consider resharper ignores as an alternative.

**Comment by peppy** [comment]
is `if (value is NonNullable resolvedValue)` a valid alternative? or just explicitly re-throwing a nullref in place of the assert `if (value == null) throw NullReferenceException(value)`?

I'm also fine with either of those (though the first one likely isn't so correct since we are implying guaranteed non-null.

**Comment by smoogipoo** [comment]
Please let me know if you're fine with the changes.

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -510,7 +512,7 @@ public void Run(Game game)
                 var assembly = DebugUtils.GetEntryAssembly();
 
                 Logger.GameIdentifier = Name;
-                Logger.VersionIdentifier = assembly.GetName().Version.ToString();
+                Logger.VersionIdentifier = assembly.GetName().Version?.ToString() ?? Logger.VersionIdentifier;
```
Comment (peppy) [inline osu.Framework/Platform/GameHost.cs:515]: not a huge fan of this one but it'll do

File: osu.Framework/Statistics/DotNetRuntimeListener.cs [hunk 2]
```diff
@@ -22,22 +22,25 @@ protected override void OnEventWritten(EventWrittenEventArgs data)
         {
             switch ((EventType)data.EventId)
             {
-                case EventType.GCStart_V1:
+                case EventType.GCStart_V1 when data.Payload != null:
```
Comment (peppy) [inline osu.Framework/Statistics/DotNetRuntimeListener.cs:25]: another thing i'm really not sure about is `when`, but given it's only in this one file i guess we can leave it for now.

**Pull request merged**